

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>NGS: SAM tools &mdash; Galaxy@CRS4 0.1 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="Galaxy@CRS4 0.1 documentation" href="../index.html" />
    <link rel="up" title="Variant Calling" href="../variant_calling.html" />
    <link rel="next" title="NGS: GATK tools" href="ngs_gatk_tools.html" />
    <link rel="prev" title="Indel analysis" href="ngs_indel_analysis/indel_analysis.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="ngs_gatk_tools.html" title="NGS: GATK tools"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="ngs_indel_analysis/indel_analysis.html" title="Indel analysis"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Galaxy@CRS4 0.1 documentation</a> &raquo;</li>
          <li><a href="../variant_calling.html" accesskey="U">Variant Calling</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="ngs-sam-tools">
<h1>NGS: SAM tools<a class="headerlink" href="#ngs-sam-tools" title="Permalink to this headline">¶</a></h1>
<p>Tools to manipulate SAM files:</p>
<dl class="docutils">
<dt><strong>Filter SAM</strong></dt>
<dd>allows parsing of SAM datasets using bitwise flag (the second column).</dd>
<dt><strong>Convert SAM</strong></dt>
<dd>converts positional information from a SAM dataset into interval format with 0-based start and 1-based end. CIGAR string of SAM format is used to compute the end coordinate.</dd>
<dt><strong>SAM-to-BAM</strong></dt>
<dd>produces an indexed BAM file based on a sorted input SAM file.</dd>
<dt><strong>BAM-to-SAM</strong></dt>
<dd>produces a SAM file from a BAM file.</dd>
<dt><strong>Merge BAM files</strong></dt>
<dd>uses the Picard merge command to merge any number of BAM files together into one BAM file while preserving the BAM metadata such as read groups.</dd>
<dt><strong>MPileup SNP and indel caller</strong></dt>
<dd>generates BCF or pileup for one or multiple BAM files. Alignment records are grouped by sample identifiers in &#64;RG header lines. If sample identifiers are absent, each input file is regarded as one sample.</dd>
<dt><strong>Generate pileup</strong></dt>
<dd>uses SAMTools&#8217; pileup command to produce a pileup dataset from a provided BAM dataset. It generates two types of pileup datasets depending on the specified options.</dd>
<dt><strong>Filter pileup on coverage and SNPs</strong></dt>
<dd>allows one to find sequence variants and/or sites covered by a specified number of reads with bases above a set quality threshold. The tool works on six and ten column pileup formats produced with SAMTools pileup command.</dd>
<dt><strong>Pileup-to-Interval</strong></dt>
<dd>reduces the size of a results set by taking a pileup file and producing a condensed version showing consecutive sequences of bases meeting coverage criteria. The tool works on six and ten column pileup formats produced with SAMTools pileup command.</dd>
<dt><strong>flagstat</strong></dt>
<dd>uses the SAMTools toolkit to produce simple stats on a BAM file.</dd>
<dt><strong>rmdup</strong></dt>
<dd>removes potential PCR duplicates: if multiple read pairs have identical external coordinates, only retain the pair with highest mapping quality. In the paired-end mode, this command ONLY works with FR orientation and requires ISIZE is correctly set. It does not work for unpaired reads (e.g. two ends mapped to different chromosomes or orphan reads).</dd>
<dt><strong>Slice BAM</strong></dt>
<dd>accepts an input BAM file and an input BED file and creates an output BAM file containing only those alignments that overlap the provided BED intervals.</dd>
<dt><strong>From SAM to Consensus/Variant</strong></dt>
<dd>performs an entire pipeline from mapping (in SAM) to the extraction of a draft consensus sequence It also calculate SNPs and shor indels by Samtools and extracts contigs for downstream analysis.</dd>
<dt><strong>Get Mapping Sequences from SAM</strong></dt>
<dd>retrieves all reads that did match the reference. It writes them in a fastQ format. It also returns a SAM file without un-mapped reads.</dd>
<dt><strong>Get UnMapped Sequences from SAM</strong></dt>
<dd>retrieves all reads that did not match the reference. It writes them in a FASTQ format.</dd>
</dl>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<div class="line-block">
<div class="line"><em>The Sequence Alignment/Map format and SAMtools</em>.</div>
<div class="line">H. Li, B. Handsaker, A. Wysoker, T. Fennell, J. Ruan, N. Homer, G. Marth, G. Abecasis, R. Durbin and 1000 Genome Project Data Processing Subgroup.</div>
<div class="line">Bioinformatics (2009) 25 (16): 2078-2079.</div>
<div class="line"><a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/25/16/2078">http://bioinformatics.oxfordjournals.org/content/25/16/2078</a></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">NGS: SAM tools</a><ul>
<li><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="ngs_indel_analysis/indel_analysis.html"
                        title="previous chapter">Indel analysis</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="ngs_gatk_tools.html"
                        title="next chapter">NGS: GATK tools</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/variant_calling/ngs_sam_tools.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="ngs_gatk_tools.html" title="NGS: GATK tools"
             >next</a> |</li>
        <li class="right" >
          <a href="ngs_indel_analysis/indel_analysis.html" title="Indel analysis"
             >previous</a> |</li>
        <li><a href="../index.html">Galaxy@CRS4 0.1 documentation</a> &raquo;</li>
          <li><a href="../variant_calling.html" >Variant Calling</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, CRS4 Bioinformatica.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>